src/reporting/file-issue-for-review.js

```diff
@@ -0,0 +1,146 @@
+const path = require("path");
+const fs = require("fs").promises;
+const { execSync } = require('child_process');
+const Octokit = require("../lib/octokit");
+
+const GH_TOKEN = (() => {
+  try {
+    return require("../config.json").GH_TOKEN;
+  } catch (err) {
+    return process.env.GH_TOKEN;
+  }
+})();
+
+const MAX_PR_BY_RUNS = 10;
+
+const repoOwner = 'w3c';
+const repoName = 'strudy';
+
+if (!GH_TOKEN) {
+  console.error("GH_TOKEN must be set to some personal access token as an env variable or in a config.json file");
+  process.exit(1);
+}
+
+const octokit = new Octokit({
+  auth: GH_TOKEN,
+  //log: console
+});
+
+function prWrapper(title, uri, repo, issueReport) {
+  return `This pull request was automatically created by strudy upon detecting errors in ${title}.
+
+Please check that these errors were correctly detected, and that they have not already been reported in ${repo}.
+
+If everything is OK, you can merge this pull request which will report the issue below to the repo, and update the underlying report file with a link to the said issue.
+
+---
+
+${issueReport}
+`;
+}
+
+if (require.main === module) {
+  let edCrawlResultsPath = process.argv[2];
+  let trCrawlResultsPath = process.argv[3];
+  // Target the index file if needed
+  if (!edCrawlResultsPath.endsWith('index.json')) {
+    edCrawlResultsPath = path.join(edCrawlResultsPath, 'index.json');
+  }
+  if (!trCrawlResultsPath.endsWith('index.json')) {
+    trCrawlResultsPath = path.join(trCrawlResultsPath, 'index.json');
+  }
+  (async function() {
+    console.log(`Opening crawl results ${edCrawlResultsPath} and ${trCrawlResultsPath}…`)
+    const crawl = await loadCrawlResults(edCrawlResultsPath, trCrawlResultsPath);
+    console.log("- done");
+    console.log("Running broken link detection…");
+    const results = await studyBackrefs(crawl.ed, crawl.tr);
+    console.log("- done");
+    const currentBranch = execSync('git branch --show-current', { encoding: 'utf8' }).trim();
+    let counter = 0;
+    const needsPush = {};
+    await Promise.all(Object.keys(results).map(async uri => {
+      const specResult = results[uri];
+      console.log(`Compiling report for ${specResult.title}…`);
+      const brokenLinks = specResult.brokenLinks || [];
+      if (brokenLinks.length) {
+
+	if (counter > MAX_PR_BY_RUNS) return;
+	const issueMoniker = `${specResult.shortname}-brokenlinks`;
+	// is there already a file with that moniker?
+	const issueFilename = path.join('issues/', issueMoniker + '.md');
+	try {
+	  if (!(await fs.stat(issueFilename)).isFile()) {
+	    console.error(`${issueFilename} already exists but is not a file`);
+	  } else {
+	    console.log(`${issueFilename} already exists, bailing`);
+	  }
+	  return;
+	} catch (err) {
+	  // Intentionally blank
+	}
+	// if not, is there a pull request that uses that moniker as a branch
+	const {data: pullrequests} = (await octokit.rest.pulls.list({
```

Comment:
I would try to keep the logic that runs locally separate from the logic that interacts with GitHub, to ease debugging of errors as interaction with GitHub can fail at any time.

Typically, I would rather check here that there is no local branch with the same moniker (regardless of the existence of a PR, the `git checkout -b` command will fail if the branch already exists locally), and only worry about what's in the GitHub repository in the final loop on `needsPush` (the `git push` command will fail in any case if the branch already exists on GitHub).

Reply:
the fact that the checkout -b will fail is the reason I'm not bothering checking locally; in any case, a local check assumes the local clone has all the branches, which may or may not be true. But you're right the PR existence check would be more logically-grouped with the PR creation - I'll move it there